packages/amplify-graphql-api-construct/src/__tests__/__functional__/auth-modes.test.ts

```diff
@@ -121,6 +122,100 @@ describe('auth modes', () => {
     });
   });
 
+  it('renders with user pool auth across data sources: sql primary, ddb related', () => {
+    const ddbSchema = /* GraphQL */ `
+      type RelatedMany @model @auth(rules: [{ allow: public, operations: [read] }, { allow: owner }]) {
+        id: String! @primaryKey
+        secret: String @auth(rules: [{ allow: owner }])
+        owner: String
+        primaryId: String
+        primary: Primary @belongsTo(references: ["primaryId"])
+      }
+
+      type RelatedOne @model @auth(rules: [{ allow: public, operations: [read] }, { allow: owner }]) {
+        id: String! @primaryKey
+        secret: String @auth(rules: [{ allow: owner }])
+        owner: String
+        primaryId: String
+        primary: Primary @belongsTo(references: ["primaryId"])
+      }
+    `;
+
+    const sqlSchema = /* GraphQL */ `
+      type Primary @model @auth(rules: [{ allow: public, operations: [read] }, { allow: owner }]) {
+        id: String! @primaryKey
+        secret: String @auth(rules: [{ allow: owner }])
+        owner: String
+        relatedMany: [RelatedMany] @hasMany(references: ["primaryId"])
+        relatedOne: RelatedOne @hasOne(references: ["primaryId"])
+      }
+    `;
+
+    const sqlstrategy1 = mockSqlDataSourceStrategy({ name: 'sqlstrategy1' });
+
+    verifySynth((stack) => {
+      const ddbDefinition = AmplifyGraphqlDefinition.fromString(ddbSchema);
+      const sqlDefinition = AmplifyGraphqlDefinition.fromString(sqlSchema, sqlstrategy1);
+      const userPool = cognito.UserPool.fromUserPoolId(stack, 'ImportedUserPool', 'ImportedUserPoolId');
+
+      new AmplifyGraphqlApi(stack, 'TestApi', {
+        definition: AmplifyGraphqlDefinition.combine([ddbDefinition, sqlDefinition]),
+        authorizationModes: {
+          defaultAuthorizationMode: 'AMAZON_COGNITO_USER_POOLS',
+          apiKeyConfig: { expires: cdk.Duration.days(7) },
+          userPoolConfig: { userPool },
+        },
+      });
+    });
+  });
+
+  it('renders with user pool auth across data sources: ddb primary, sql related', () => {
```

Comment:
It probably shouldn't matter which side of the relationship has which data source, but better safe than sorry